lib/rules/keyframe-selector-notation/index.mjs

```diff
@@ -101,27 +95,56 @@ const rule = (primary) => {
 						return;
 					}
 
-					if (expFunc(selector, selectorsInBlock)) return;
+					if (expFunc(normalizedSelector, selectorsInBlock)) return;
 
-					const fixedSelector = fixFunc(selector);
+					const fixedSelector = fixFunc(normalizedSelector);
 
 					report({
 						message: messages.expected,
-						messageArgs: [selector, fixedSelector],
+						messageArgs: [keyframeSelector.value, fixedSelector],
 						node: keyframeRule,
 						result,
 						ruleName,
-						word: selector,
+						index: keyframeSelector.sourceIndex,
+						endIndex: keyframeSelector.sourceIndex + normalizedSelector.length,
 						fix: () => {
-							tag.value = fixedSelector;
+							keyframeSelector.value = fixedSelector;
+							keyframeRule.selector = selectors.toString();
 						},
 					});
-				}
+				});
 			});
 		});
 	};
 };
 
+/** @import { Selector, Tag } from 'postcss-selector-parser' */
+
+/**
+ * Full syntax is:
+ * `<keyframe-selector> = from | to | <percentage [0,100]> | <timeline-range-name> <percentage>`
+ *
+ * Only parses `from | to | <percentage [0,100]>` and returns `undefined` in any other case.
```

Comment:
[suggestion] From this comment, how about checking `keyframeSelectorKeywords` and `PERCENTAGE_SELECTORS` inside this `parseKeyframeSelector` function? If this idea makes sense, `parseKeyframeSelector()` would return two values: `keyframeSelector` and `normalizedSelector`, though.

Reply:
I agree that it makes sense to group these checks.